src/App.tsx

```diff
@@ -1,204 +1,117 @@
-/* eslint-disable jsx-a11y/label-has-associated-control */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import * as todosApi from './api/todos';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { getFilteredTodos } from './utils/getFilterTodos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { wait } from './utils/fetchClient';
+import { Error } from './components/Error';
+
+const getNextTodoId = (todo: Todo[]) => {
+  const ids = todo.map(item => item.id);
+
+  return Math.max(...ids, 0) + 1;
+};
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [filter, setFilter] = useState<Status>(Status.All);
+
+  useEffect(() => {
+    todosApi
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(`Unable to load todos`);
+        wait(3000).then(() => {
+          setErrorMessage('');
+        });
+      });
+  }, []);
+
+  function addTodo({ title, completed, userId }: Todo) {
+    todosApi
+      .createTodos({ title, completed, userId })
+      .then(newTodo => {
+        setTodos(currentTodos => {
+          return [...currentTodos, newTodo];
+        });
+      });
+  }
+
+  const visibleTodos = getFilteredTodos([...todos], filter);
+
+  const handleTitleChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodoTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    addTodo({
+      id: getNextTodoId(todos),
+      title: todoTitle,
+      completed: false,
+      userId: getNextTodoId(todos),
+    });
+
+    setTodoTitle('');
+  };
+
+  if (!todosApi.USER_ID) {
     return <UserWarning />;
   }
 
+  const itemLeft = (items: Todo[]): number => {
+    const item = items.filter(i => !i.completed);
+
+    return item.length;
+  };
```

Comment:
- Name looks like it is a variable, but actually it's a function
- Avoid one-letter arguments/variables
- Move this function to utils